### PR TITLE
DRY handshake code

### LINKED ISSUE #422: DRY connection/channel updates
Now that `chain.UpdateClient` does all updating functionality. I'm pretty sure we can reduce several lines of code by calling src.UpdateClient and dst.UpdateClient in the beginning of connection/channel handshakes and using those messages when necessary. To fix this issue, simply construct the update message for src and dst once, at the top of ExecuteConnection(Channel)Handshake, and pass those into Initialize, and use the constructed messages instead of creating one in each switch statement

### PR DESCRIPTION
**Pull request by colin-axner**
closes #422 

I believe this fixes a bug where the handshake proof relies on a consensus state not being committed via an update message

### CODE REVIEW

**Comment by colin-axner** [comment]
a more comprehensive fix should be done in #425

File: relayer/naive-strategy.go [hunk 1]
```diff
@@ -563,7 +562,7 @@ func (nrs *NaiveStrategy) RelayPackets(src, dst *Chain, sp *RelaySequences) erro
 	}
 
 	if len(msgs.Src) != 0 {
-		updateMsg, err := dst.UpdateClient(src)
+		updateMsg, err := src.UpdateClient(dst)
```
Comment (AdityaSripal) [inline relayer/naive-strategy.go:565]: Was this wrong before?

**Review by AdityaSripal — APPROVED** [review]
utACK

File: relayer/naive-strategy.go (hunk 1 above)
Comment (colin-axner) [reply relayer/naive-strategy.go:565]: yes pretty sure. The logic doesn't hold since if you look at the code, this message is beings added to msgs.Src

**Pull request merged**